tests/e2e/e2e_test.go

```diff
@@ -137,16 +126,44 @@ func (s *IntegrationTestSuite) Test03SuperfluidVoting() {
 	)
 }
 
+// TestAddToExistingLockPostUpgrade tests lockups to both regular and superfluid locks.
+// Specifically, we ensure addToExistingLock works for both preupgrade and postupgrade locks.
+func (s *IntegrationTestSuite) TestAddToExistingLockPostUpgrade() {
+	chainA := s.configurer.GetChainConfig(0)
+	chainANode, err := chainA.GetDefaultNode()
+	s.NoError(err)
+	if !s.skipUpgrade {
+		// ensure we can add to existing locks and superfluid locks that existed pre upgrade on chainA
+		// we use the hardcoded gamm/pool/1 as this was the one that was created pre upgrade
+		lockupWalletAddrA, lockupWalletSuperfluidAddrA := chainANode.GetWallet("lockup-wallet"), chainANode.GetWallet("lockup-wallet-superfluid")
+		chainANode.AddToExistingLock(sdk.NewInt(1000000000000000000), "gamm/pool/1", "240s", lockupWalletAddrA)
+		chainANode.AddToExistingLock(sdk.NewInt(1000000000000000000), "gamm/pool/1", "240s", lockupWalletSuperfluidAddrA)
+	}
+
+	// ensure we can add to new locks and superfluid locks
+	// create pool and enable superfluid assets
+	poolId := chainANode.CreatePool("nativeDenomPool.json", chainA.NodeConfigs[0].PublicAddress)
+	chainA.EnableSuperfluidAsset(fmt.Sprintf("gamm/pool/%d", poolId))
+
+	// setup wallets and send gamm tokens to these wallets on chainB
+	lockupWalletAddrB, lockupWalletSuperfluidAddrB := chainANode.CreateWallet("TestAddToExistingLockPostUpgrade"), chainANode.CreateWallet("TestAddToExistingLockPostUpgradeSuperfluid")
```

Comment:
If we go with the splitting suggestion above, it might make sense to rename walltes as well